### PR TITLE
Channel manager: sql dao implementation

### PR DESCRIPTION
**Pull request by lindlind**

### CODE REVIEW

File: lzy/channel-manager/src/main/java/ai/lzy/channelmanager/dao/ChannelDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,515 @@
...
+                st.setString(++index, toJson(channelSpec));
+
+                st.setString(++index, Channel.LifeStatus.ALIVE.name());
+                st.setTimestamp(++index, Timestamp.from(createdAt.truncatedTo(ChronoUnit.MILLIS)));
+                st.setTimestamp(++index, Timestamp.from(createdAt.truncatedTo(ChronoUnit.MILLIS)));
+
+                try {
+                    st.executeUpdate();
+                } catch (SQLException e) {
+                    if (e.getMessage().contains("channels_pkey")) {
```
Comment (imakunin) [inline lzy/channel-manager/src/main/java/ai/lzy/channelmanager/dao/ChannelDaoImpl.java:154]: `DbHelper::isUniqueViolation` ?

Comment (imakunin) [reply lzy/channel-manager/src/main/java/ai/lzy/channelmanager/dao/ChannelDaoImpl.java:154]: or
```
  NOT_NULL_VIOLATION("23502"),
  FOREIGN_KEY_VIOLATION("23503"),
  UNIQUE_VIOLATION("23505"),
  CHECK_VIOLATION("23514"),
  EXCLUSION_VIOLATION("23P01"),
```

File: lzy/channel-manager/src/main/java/ai/lzy/channelmanager/test/BaseTestWithChannelManager.java [hunk 2]
```diff
@@ -42,7 +40,6 @@ public void after() throws InterruptedException {
         }
         app.stop();
         app.awaitTermination();
-        DatabaseTestUtils.cleanup(context.getBean(ChannelManagerDataSource.class));
```
Comment (imakunin) [inline lzy/channel-manager/src/main/java/ai/lzy/channelmanager/test/BaseTestWithChannelManager.java:45]: replace it with
```
context.getBean(ChannelManagerDataSource.class).setOnClose(DatabaseTestUtils::cleanup);
```

File: lzy/channel-manager/src/main/resources/db/channel-manager/migrations/V1_0__Init.sql [hunk 3]
```diff
@@ -33,3 +33,73 @@ CREATE TABLE channel_operation
...
+
+CREATE UNIQUE INDEX channels_execution_id_channel_name_idx
+    ON channels (execution_id, channel_name);
+
+CREATE TYPE endpoint_life_status_type AS ENUM ('BINDING', 'BOUND', 'UNBINDING');
+
+CREATE TABLE endpoints
+(
+    slot_uri    TEXT                      NOT NULL,
+    "slot_name" TEXT                      NOT NULL,
```
Comment (imakunin) [inline lzy/channel-manager/src/main/resources/db/channel-manager/migrations/V1_0__Init.sql:63]: why do you quote this column name?

Comment (lindlind) [reply lzy/channel-manager/src/main/resources/db/channel-manager/migrations/V1_0__Init.sql:63]: It's highlighted with black bold font like a keyword without quotes.

**Review by imakunin — APPROVED** [review]

**Pull request merged**